code/modules/reagents/chemistry/holder.dm

```diff
@@ -233,6 +234,15 @@ var/const/INGEST = 2
 	if(M)
 		temperature_reagents(M.bodytemperature - 30)
 
+
+	if(LAZYLEN(addiction_threshold_accumulated))
```

Comment:
Is this check needed? The for won't do anything if the list is empty. 
Unless this is for performance reasons ofc

Reply:
No point in really running the below with an empty list.